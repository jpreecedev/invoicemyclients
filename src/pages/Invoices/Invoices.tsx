import React from 'react'

import { ROUTES } from '../../routes'
import { AuthenticatedPage } from '../../components/AuthenticatedPage'

const InvoicesPage = () => {
  return (
    <AuthenticatedPage>
      <a href={ROUTES.invoiceBuilder}>Invoice Builder</a>
    </AuthenticatedPage>
  )
}

export { InvoicesPage }
