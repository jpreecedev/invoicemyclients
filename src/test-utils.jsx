import React from 'react'
import { MemoryRouter as Router } from 'react-router-dom'
import { render } from '@testing-library/react'
import { createMemoryHistory } from 'history'

function renderWithRouter(
  ui,
  { route = '/', history = createMemoryHistory({ initialEntries: [route] }) } = {}
) {
  const Wrapper = ({ children }) => <Router history={history}>{children}</Router>
  return {
    ...render(ui, { wrapper: Wrapper }),
    // adding `history` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    history
  }
}

export { renderWithRouter }
