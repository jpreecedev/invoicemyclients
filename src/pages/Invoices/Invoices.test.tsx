import React from 'react'
import { renderWithRouter } from '../../test-utils'

import { InvoicesPage } from './Invoices'

describe('<Invoices /> tests', () => {
  it('should match snapshot', () => {
    expect(renderWithRouter(<InvoicesPage />).container).toMatchSnapshot()
  })
})
