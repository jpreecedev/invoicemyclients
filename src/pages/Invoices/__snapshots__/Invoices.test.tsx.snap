// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Invoices /> tests should match snapshot 1`] = `
<div>
  <nav
    class="navbar navbar-expand-lg navbar-light"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        InvoiceMyClients.com
      </a>
      <button
        aria-controls="navbarCollapse"
        aria-expanded="false"
        aria-label="Toggle navigation"
        class="navbar-toggler"
        data-target="#navbarCollapse"
        data-toggle="collapse"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
        id="navbarCollapse"
      >
        <button
          aria-controls="navbarCollapse"
          aria-expanded="false"
          aria-label="Toggle navigation"
          class="navbar-toggler"
          data-target="#navbarCollapse"
          data-toggle="collapse"
          type="button"
        >
          <i
            class="fe fe-x"
          />
        </button>
        <ul
          class="navbar-nav ml-auto"
        >
          <li
            class="nav-item dropdown"
          >
            <a
              class="nav-link"
              href="/dashboard"
            >
              Dashboard
            </a>
          </li>
          <li
            class="nav-item dropdown"
          >
            <a
              class="nav-link"
              href="/clients/list"
            >
              Clients
            </a>
          </li>
          <li
            class="nav-item dropdown"
          >
            <a
              class="nav-link"
              href="/invoices"
            >
              Invoices
            </a>
          </li>
          <li
            class="nav-item dropdown"
          >
            <a
              class="nav-link"
              href="/nagging"
            >
              Nagging
            </a>
          </li>
          <li
            class="nav-item dropdown"
          >
            <a
              class="nav-link"
              href="/support"
            >
              Support
            </a>
          </li>
        </ul>
        <a
          class="navbar-btn btn btn-sm btn-primary lift ml-auto"
          href="/sign-up"
        >
          Sign Up
        </a>
      </div>
    </div>
  </nav>
  <main
    class="mt-5 mb-10 pb-10"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-12"
        >
          <h1>
            Invoices
          </h1>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-12"
        >
          <a
            class="btn btn-primary d-inline-flex"
            href="/invoices/add"
          >
            Invoice Builder
          </a>
        </div>
      </div>
    </div>
  </main>
</div>
`;
